core/Objects/NPMClient.cs

```diff
@@ -146,6 +171,204 @@ public IPackage[] FindPlugins(Uri[] registries)
+            Log.Warning($"Downloading package {pkg.Dist.Tarball}");
+            // Get the HTTP response as a stream
+            using var response = await client.GetAsync(pkg.Dist.Tarball, HttpCompletionOption.ResponseHeadersRead);
+            response.EnsureSuccessStatusCode(); // Throw if not successful
+            Log.Debug("Download got a successful code, saving to temp file");
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+            // Open the content stream from the HTTP response
+            await using var contentStream = await response.Content.ReadAsStreamAsync();
+            Log.Debug("Got response");
+            // Stream the content directly into the file
+            await contentStream.CopyToAsync(fileStream);
+            Log.Debug($"Saving to file {output.FullName} finished, took {stopwatch.ElapsedMilliseconds}ms");
+            return output;
+        }
+
+        public async Task PublishPackage(IFileInfo package)
+        {
+            var toLowerCase = true; // TODO: complete implementation
+           
+            var ctrlr = new CmfPackageController(package, package.FileSystem);
+            
+            var manifest = ctrlr.ToJson(toLowerCase);
+
+            var name = toLowerCase ? ctrlr.CmfPackage.PackageId.ToLowerInvariant() : ctrlr.CmfPackage.PackageId;
+            var tgz = $"{name}-{ctrlr.CmfPackage.Version}.tgz";
+            Log.Debug($"Trying to publish {name} to {this.baseUrl}");
+            JObject root = null;
+            
+            try
+            {
+                Log.Debug("Load package content...");
+                using var fileStream = package.OpenRead();
+                using var memoryStream = new MemoryStream();
+                
+                fileStream.CopyTo(memoryStream);  // this can lead to large memory consumption
+                byte[] fileBytes = memoryStream.ToArray();
+
+                string dataBase64 = Convert.ToBase64String(fileBytes);
+
+                using SHA1 sha1 = SHA1.Create();
+                byte[] sha1HashBytes = sha1.ComputeHash(fileBytes);
+                var sha1Hash = BitConverter.ToString(sha1HashBytes).Replace("-", "").ToLower();
+
+                using SHA512 sha512 = SHA512.Create();
+                byte[] sha512HashBytes = sha512.ComputeHash(fileBytes);
+                var sha512_64 = Convert.ToBase64String(sha512HashBytes);
+                // var sha512Hash = BitConverter.ToString(sha512HashBytes).Replace("-", "").ToLower();
+                var sha512Hash = $"sha512-{sha512_64}";
+                Log.Debug($"Package content with hash {sha512Hash} and checksum {sha1Hash}");
+                
+                root = JObject.Parse(
+                    $$"""
+                    { 
+                        "_id": "{{name}}",
+                        "name": "{{name}}",
+                        "description": "{{ctrlr.CmfPackage.Description}}",
```

Comment:
Isn't creating the JSON this way risky? I know the name is unlikely to have double quotes, but the description might have, and it would cause it to break.